bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/servlets/AdaptiveImageServlet.java

```diff
@@ -339,11 +339,8 @@ private void transformAndStreamAsset(SlingHttpServletResponse response, ValueMap
             if (!appliedTransformation) {
                 EnhancedRendition rendition = getBestRendition(asset, resizeWidth);
                 Dimension dimension = rendition.getDimension();
-                if (dimension != null) {
-                    originalWidth = dimension.width;
-                    originalHeight = dimension.height;
-                }
-                if (originalHeight > resizeWidth) {
+                if ((dimension != null && dimension.width > resizeWidth) ||
+                    (dimension == null && originalWidth > resizeWidth)) {
                     int resizeHeight = calculateResizeHeight(originalWidth, originalHeight, resizeWidth);
                     if (resizeHeight > 0 && resizeHeight != originalHeight) {
```

Comment:
The case where `resizeHeight == 0` is currently missing because of the added if statement

Reply:
That's another issue than the aspect ration but well spoted!!!